standards/SPIFFE_Federation.md

```diff
@@ -0,0 +1,316 @@
+Trust Domain:			"prod.example.com"
+```  
+*Figure 1: Example SPIFFE bundle endpoint configuration for trust domain prod.example.com. Administrators configure SPIFFE control planes to retrieve foreign trust bundles through a bundle endpoint configuration.*
+
+When a control plane internally distributes trust bundles to workloads, the association between trust domain name and trust bundle MUST be communicated. Please see the [Security Considerations](#7-security-considerations) section for more information about the sensitivity of these parameters.
+
+![SPIFFE bundle distribution](https://raw.githubusercontent.com/evan2645/spiffe/ecb9fb894e1a13fae821c370f7b5ddeee634d1c8/standards/img/spiffe_bundle_distribution.png)  
+*Figure 2: After retrieving a foreign SPIFFE trust bundle, control planes distribute both the trust domain name and the corresponding bundle to internal workloads. Workloads use this configuration to validate identities in the foreign trust domain. For details about the trust bundle contents, refer to The [SPIFFE Trust Domain and Bundle][3], specifically the [SPIFFE Bundle Format][6] and [SPIFFE Bundle Example][4] sections.*
+
+The requirements in this section apply to all SPIFFE bundle endpoint servers and clients. Individual SPIFFE bundle endpoint profiles MAY add further requirements.
+
+### 5.2. Endpoint Profiles
+An endpoint profile describes both the transport protocol and authentication method that should be used when serving or consuming a bundle endpoint.
+
+The following sections describe the supported bundle endpoint profiles.
+
+#### 5.2.1. Web PKI (`https_web`)
+The `https_web` profile leverages publicly trusted certificate authorities to provide a low-friction path for configuring SPIFFE federation. It behaves identically to the "https" URLs that most people are familiar with when accessing web pages in their browsers. In this profile, the bundle endpoint server uses a certificate issued by a public CA, obviating the need for additional client configuration; endpoints using the `https_web` profile type are authenticated using the same public CA certificates commonly installed in modern operating systems.
+
+Please see the [Security Considerations](#7-security-considerations) section for more information about the use of public certificate authorities.
+
+##### 5.2.1.1. Endpoint URL Requirements
+Bundle endpoint URLs utilizing `https_web` MUST have the scheme set to `https` and MUST NOT include userinfo in the authority component. Other components of the URL (as defined by [RFC 3986 section 3][7]) are not constrained by this specification.
+
+For example, the URL `https://host.example.com/trust_domain` is a valid SPIFFE bundle endpoint URL for the `https_web` profile type.
+
+##### 5.2.1.2. Endpoint Parameters
+The `https_web` profile MUST NOT require any additional parameters beyond those required for every profile (namely the trust domain name, profile type, and endpoint URL) in order to function.
```

Comment:
I'm now wondering if this requirement is too strict.  Imagine a client written in a fully-portable framework that doesn't have access to the operating system's root CA store (or being deployed on an operating system that doesn't have a root CA store at all), and thus requires the CA store to be specified as a mandatory parameter.  Would we be willing to call that implementation non-complying?

Reply:
I think that in your example, the software as a whole would be configured with a path to the CA store (as opposed to being defined on a per-endpoint basis)

Reply:
That's true.

More generally where my mind is going is that we've made an assumption that given the endpoint URL, the client will always have all the information it needs to access a particular endpoint server and validate it.  Does that assumption hold in all cases?  Let's think networking: could an implementation insist that you configure the network interface, or resolver, or IP version, or... ?  For example, software that gets deployed on a gateway between two networks with different security classifications---you might insist that each endpoint gets marked with the appropriate network to prevent DNS queries of secret names being executed on unclass networks.

Maybe such a scenario is too exotic to worry about; WDYT?

Reply:
Hmm... that is a good question...

IMO, a big part of this requirement is interop as it relates to the advertisement of endpoint parameters. A custom implementation may require such details (as illustrated in your most recent example), but it's kind of bespoke at that point? Can you say that such an implementation is conformant? With the current text, the answer is no... but does that person care? They'll certainly tell others that they are conformant!

We've had at least one community request for some rigidity around the way that federation relationships are configured such that it is easy to distribute coordinates or parameters. While codifying an actual format by which this occurs feels out-of-scope for this PR, it _does_ feel valuable to align on what the contents of such a document will be. So maybe the real question here is, do we want to try to preserve this property or not?